pkg/build/gobuild.go

```diff
@@ -274,6 +278,63 @@ func getGoBinary() string {
 	return defaultGoBin
 }
 
+func getDelve(ctx context.Context, platform v1.Platform) (string, error) {
+	env, err := buildEnv(platform, os.Environ(), nil, nil)
+	if err != nil {
+		return "", fmt.Errorf("could not create env for Delve build: %w", err)
+	}
+
+	tmpInstallDir, err := os.MkdirTemp("", "delve")
+	if err != nil {
+		return "", fmt.Errorf("could not create tmp dir for Delve installation: %w", err)
+	}
+
+	// install delve to tmp directory
+	env = append(env, fmt.Sprintf("GOPATH=%s", tmpInstallDir))
+
+	args := []string{
+		"install",
+		"github.com/go-delve/delve/cmd/dlv@latest",
+	}
+
+	gobin := getGoBinary()
+	cmd := exec.CommandContext(ctx, gobin, args...)
+	cmd.Env = env
+
+	var output bytes.Buffer
+	cmd.Stderr = &output
+	cmd.Stdout = &output
+
+	log.Printf("Building Delve for %s", platform)
+	if err := cmd.Run(); err != nil {
+		os.RemoveAll(tmpInstallDir)
+		return "", fmt.Errorf("go build Delve: %w: %s", err, output.String())
+	}
+
+	// find the delve binary in tmpInstallDir/bin/
+	delveBinary := ""
+	err = filepath.WalkDir(filepath.Join(tmpInstallDir, "bin"), func(path string, d fs.DirEntry, err error) error {
```

Comment:
Why do we have to walk? Don't we know it'll be at `tmpInstallDir/bin/dlv`?

Reply:
That's a good point... We probably want to confirm the end state after the "go install", but IIUC we could do that more directly with a stat call with the expected path.

Reply:
Ah I just hit a snag here that I'm guessing motivated this design: testing again locally, the binary was at `tmpInstallDir/bin/linux_arm64/dlv`.

Reply:
Hey @imjasonh,
the motivation behind this is as @luhring said in https://github.com/ko-build/ko/pull/1320#discussion_r1607491196: for other archs it can be located at different locations/folders.

Reply:
That makes sense! In that case, should we only look at `tmpInstallDir/bin/$GOOS_GOARCH/dlv`? I'm worried the walking might open us up to a bug where `dlv` gets installed in the image for the wrong arch.

Reply:
Pushed e1438e2 to address this, please let me know if that's looking any better!